connectors/sources/sharepoint_online.py

```diff
@@ -2182,6 +2224,18 @@ def update_drive_delta_link(self, drive_id, link):
     def get_drive_delta_link(self, drive_id):
         return self._sync_cursor.get(CURSOR_SITE_DRIVE_KEY, {}).get(drive_id)
 
+    def last_sync_time(self):
+        return self._sync_cursor.get(CURSOR_SYNC_TIMESTAMP, self.epoch_timestamp())
+
+    def current_timestamp(self):
+        return datetime.now().strftime(TIMESTAMP_FORMAT)
+
+    def epoch_timestamp(self):
+        return strftime(TIMESTAMP_FORMAT, time.gmtime(0))
+
+    def update_sync_timestamp_cursor(self, timestamp):
+        self._sync_cursor[CURSOR_SYNC_TIMESTAMP] = timestamp
```

Comment:
These methods look like good candidates for moving them to the framework (outside of the PR scope, but probably worth creating an issue).